test/tests/performance/pbs_node_buckets_perf.py

```diff
@@ -65,13 +65,43 @@ def cust_attr_func(self, name, totalnodes, numnode, attribs):
         """
         a = {'resources_available.color': self.colors[numnode % 7]}
         return dict(attribs.items() + a.items())
+    
+    def submit_jobs(self, attribs, num):
+        """
+        Submit num jobs each in their individual equiv class
+        """
+        wt = 100
+        jids = []
+        for i in range(num):
+            attribs['Resource_List.walltime'] = wt + i
+            J = Job(TEST_USER, attrs=attribs)
+            jid = self.server.submit(J)
+            jids.append(jid)
+            
+        return jids
+    
+    def run_cycle(self):
+        """
+        Run a cycle and return the length of the cycle
+        """
+        t = int(time.time())
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'True'})
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'False'}, expect=True)
+        
+        self.scheduler.log_match('Leaving Scheduling Cycle', starttime=t,
+                                 max_attempts=240, interval=2)
+        c = self.scheduler.cycles(lastN=1)[0]
+        return c.end - c.start
 
     @timeout(10000)
-    def test_node_bucket_perf(self):
+    def test_node_bucket_perf_scatter(self):
         """
         Submit a large number of jobs which use node buckets.  Run a cycle and
         compare that to a cycle that doesn't use node buckets.  Jobs require
-        place=scatter:excl to use node buckets.
+        place=excl to use node buckets.
+        This test uses place=scatter.  Scatter placement is quicker than free
         """
         num_jobs = 1000
         a = {'Resource_List.select': '1429:ncpus=1:color=yellow',
```

Comment:
I know this is old code, but can you please explain why you are submitting this job before proceeding with submitting the large number of jobs?

Reply:
It's to make sure the scheduler can't run any of the 1000 jobs.  We want to treat each of the jobs equally, so that means we need to either run them all or not run any.  Submitting one first means we won't run any.  I added a code comment.